cpp/src/reductions/hash_reduce_by_row.cuh

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```

Comment:
This file is extracted from `distinct_reduce.cu` thus I keep the copyright as-is.